lib/src/components/NxToast/NxToastProvider.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import ToastContext from './contexts';
+import NxToastContainer from './NxToastContainer';
+import { NxToastProviderProps, nxToastProviderPropTypes, ToastAddModel, ToastModel } from './types';
+
+let id = 1;
+
+const NxToastProvider = (props: NxToastProviderProps) => {
+  const { children } = props;
+  const [toasts, setToasts] = useState<ToastModel[]>([]);
+  const [activeElementBeforeToast, setActiveElementBeforeToast] = useState<HTMLElement | null>(null);
+
+  const addToast = useCallback((content: ToastAddModel) => {
+    //Before displaying the toast, keep track of active element
+    const currentFocusedElement = document.activeElement as HTMLElement;
+
+    //When the toast is displayed, the focus is on the Toast's close button
+    //In this case, the active element is the Toast's close button, so ignore it
+    if (!currentFocusedElement.classList.contains('nx-toast__close')) {
+      setActiveElementBeforeToast(currentFocusedElement);
+    }
+
+    setToasts((toasts) => [
+      ...toasts,
+      { toastId: id++, type: content.type, message: content.message }
```

Comment:
I'd prefer if the mutation of the `id` variable was its own statement.  It's a bit hidden here, and I can also never remember which of `id++` and `++id` is which :slightly_smiling_face:.

Reply:
Ok, I changed the `id` variable in this commit: https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/ec029e1ac2eaa2552761a4e68de619c948aedd5a

Reply:
> Ok, I changed the id variable in this commit: https://github.com/sonatype/sonatype-react-shared-components/commit/ec029e1ac2eaa2552761a4e68de619c948aedd5a

That completely avoids addressing the issue I was referring to haha.  Please do not use the result of the `++` operator as an expression.  Do the `++` in one statement, and do the assignment in another.

Reply:
>That completely avoids addressing the issue I was referring to haha. Please do not use the result of the ++ operator as an expression. Do the ++ in one statement, and do the assignment in another.

Looks like @entroform fixed this issue in this commit: https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/778629a1c81bdc8bcc3f2c58325af1389ed4ec88